libathemecore/services.c

```diff
@@ -617,6 +617,15 @@ handle_certfp(struct sourceinfo *si, struct user *u, const char *certfp)
 	if (u->myuser != NULL)
 		return;
 
+	req.si = si;
+	req.mu = mu;
```

Comment:
`mu` is uninitialized at this point

Reply:
can we just go with expecting `mu` to be NULL sometimes? unsure what that'd break

Reply:
`sasl_blacklist` in contrib needs a valid `mu` for logging purposes (it prints the account name they attempted to log into).  This is thus far the only consumer of `user_can_login` in atheme or contrib.

Reply:
see https://github.com/atheme/atheme-contrib-modules/pull/53, and incoming commit to this PR